### ISSUE TITLE
Unable to add developers to teams

### ISSUE THREAD

**Issue by arunz6161** [apigee/apigee-edge-drupal#244]
**Describe the bug**
No result shows up when you try searching for developers on the add members page for teams.

**To Reproduce**
1. Go to teams page and select an existing team.
2. Click on add members
3. Try entering the name of a developer partially or completely 
4. No developer shows up

**Expected behavior**
Autocomplete should populate developer names matching the entered data.

**Screenshots**
![Screen Shot 2019-07-24 at 3 38 16 PM](https://user-images.githubusercontent.com/4381612/61833568-8e8ce000-ae29-11e9-88d9-3aa0c0657a9f.png)

**Additional context**
None

**Version Info**
Apigee teams version: 8.x-1.0
Apigee edge version: 8.x-1.0

**Comment by arlina-espinoza** [comment]
It currently searches for a developer by username, but the dropdown will show full names (i.e. _"First Last"_), which is confusing. I think a more user-friendly approach would be to search/display developers by email. What to you think @cnovak @arunz6161 ?
Additionally, the field's help text should state what field is used for searching, and that field should be displayed in the autocomplete.
Also, just as a side note, it will search for Edge developers that are drupal users, so it would need the developer sync to be run beforehand in some scenarios.

**Comment by mxr576** [comment]
Okay, let me explain how it is working now and why.

>  I think a more user-friendly approach would be to search/display developers by email.

TL;DR; I agree that the label of the field and the description could be more clear regarding this.

> It currently searches for a developer by username, but the dropdown will show full names

It searches in email addresses and not in usernames because this is what the [List developers](https://apidocs.apigee.com/docs/management/apis/get/organizations/%7Borg_name%7D/developers) API returns when the request contains the `expand=false` query parameter. This field uses the `expand=false` parameter to only load email addresses because loading 10k expanded user objects for a simple dropdown would be really slow - especially on on-prem installations without CPS. Loading only a list of email addresses is way faster.
This explains why this field does not find matches for first names and last names. Why this field does not search in Drupal users? Well, the two main reasons are:
1) If a user has permission to manage team members (and it is not necessarily an admin user!) and tries to add a Drupal user to a team (company) that do not have a developer account on Apigee Edge then the API call fails. This happens after the form submission and it would be a bad UX communicating to a user who probably does not know anything about the API backend that this user cannot be added to the team because the Drupal user does not have a developer account. (Trying to create the missing developer is also not a good idea because it could happen that the Drupal user has invalid user data (like an email address with an unsupported email domain by Edge) and this is the reason why it could not be created before.)
2) Users who can manage team members may not have access to user profile data in Drupal.

> Also, just as a side note, it will search for Edge developers that are drupal users

Yes, the system currently filters the list of developers (email addresses) and only returns those who have Drupal users. Just because a user in Drupal has permission to manage team members it does mean that it knows anything about the MGMT API backend the system connects, it also means it does not know anything about developers on the "other side". Consequently, why would a user with this permission try to add a developer to a team who does not have a Drupal user? 

>  but the dropdown will show full names (i.e. "First Last"), 

IIRC it only shows the Drupal user's full name if the currently logged in user has access to view the user profiles and if it has this permission it makes more sense to show the full name of the selected user to him/her than the email address.

**Comment by cnovak** [comment]
Thanks for context @mxr576. @arlina-espinoza this sounds good, more help context on page and using email address for autocomplete would be clear.  

We are missing the concept of inviting users to be part of a team that are not yet registered on the portal, but we can leave that as a future enhancement.

**Comment by arlina-espinoza** [comment]
@mxr576 @cnovak The dropdown currently shows the user in the format _First Last_, without checking any sort of permissions, so again, a search by emails and displaying emails would be better.
Another thing is that by showing autocomplete suggestions to users without _"View user information"_ or _"Administer users"_ permissions might be exposing data that does not need to be shared, even if the user has the _"Manage team members"_ team permission. I think the right approach if/when we add the team invite functionality @cnovak mentioned above would be to remove the autocomplete widget and replace it with a regular textfield where a team administrator could input the email addresses, and if the mails are in the system it would add them to the team, or if not, send an invite link.

**Comment by mxr576** [comment]
> The dropdown currently shows the user in the format First Last, without checking any sort of permissions,

The TeamMembersSelection extends UserSelection which performs access check here: ` \Drupal\user\Plugin\EntityReferenceSelection\UserSelection::entityQueryAlter()`

Member invite functionality sounds good but
> would be to remove the autocomplete widget and replace it with a regular textfield 

This could be a bad UX as well. The current autocomplete solution also does not allow to add a member to a team who is already a member. A simple text field won't be able to prevent this before form validation/submission and throwing an error in form validation or ignoring any submitted value in the submit handler just because it is an existing member would not be nice. We need a more complex solution that could communicate to the user which users get an invite and which gets added immediatelly because they have a user in the system.
